bootstore/src/schemes/v0/messages.rs

```diff
@@ -0,0 +1,128 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Messages sent between peers
+
+use super::{LearnedSharePkg, Share, SharePkg};
+use derive_more::From;
+use serde::{Deserialize, Serialize};
+use sled_hardware::Baseboard;
+use uuid::Uuid;
+
+/// The first thing a peer does after connecting or accepting is to identify
+/// themselves to the connected peer.
+///
+/// This message is interpreted at the peer (network) level, and not at the FSM level,
+/// because it is used to associate IP addresses with [`Baseboard`]s.
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct Identify(Baseboard);
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Envelope {
+    pub to: Baseboard,
+    pub msg: Msg,
+}
+
+#[derive(From, Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub enum Msg {
+    Req(Request),
+    Rsp(Response),
+}
+
+impl Msg {
+    pub fn request_id(&self) -> Uuid {
+        match self {
+            Msg::Req(req) => req.id,
```

Comment:
Dumb question: should the `Uuid` move into the envelope, since it's always present?

Reply:
It's not a dumb question, but it cannot move unfortunately. Envelopes don't actually get sent over the wire. They purely exist to allow routing of messages to the network connection by the caller.